big_tests/tests/fast_auth_token_SUITE.erl

```diff
@@ -418,23 +543,68 @@ auth_with_token(Success, Token, Config, Spec, RequestToken) ->
     end,
     Res.
 
-auth_function(dont_request_token) ->
-    auth_using_token;
-auth_function(request_token) ->
-    auth_using_token_and_request_token;
-auth_function(request_invalidation) ->
-    auth_using_token_and_request_invalidation;
-auth_function(request_invalidation_1) ->
-    auth_using_token_and_request_invalidation_1.
-
-steps(success, AuthFun) ->
-    [connect_tls, start_stream_get_features,
-     {?MODULE, AuthFun},
+set_early_data(Spec, false, _Authenticate) ->
+    Spec;
+set_early_data(Spec, true, Authenticate) ->
+    Map = proplists:to_map(Spec),
+    #{ssl_opts := SslOpts} = Map,
+    Map2 = Map#{
+         ssl_opts => [{early_data, make_early_data(Spec, Authenticate)} | SslOpts]},
+    proplists:from_map(Map2).
+
+make_early_data(Spec, Authenticate) ->
+    Start = start_stream(Spec),
+    Auth = exml:to_binary(Authenticate),
+    <<Start/binary, Auth/binary>>.
```

Comment:
Huh, `ssl` should have accepted here iodata, might have been more performant. Anyway, you could do to_binary just once, by
```erlang
start_stream(Spec) ->
    escalus_tcp:stream_start_req(Spec)).

make_early_data(Spec, Authenticate) ->
    Start = start_stream(Spec),
    exml:to_binary([Start, Authenticate]).
```

But anyway, these are tests, perf is not that critical.